### PR TITLE
WebServer: Add optional listen address argument

### PR DESCRIPTION
**Pull request by EdwinHoksberg**
With this we can make the Webserver listen on another address than the default of `0.0.0.0`. If we for example pass `127.0.0.1` as the listen address, the WebServer can only be reached from within Serenity itself and not from outside hosts.

This is my first contribution so I picked a small feature to contribute, I hope I did everything okay :)

### CODE REVIEW

File: Userland/Services/WebServer/main.cpp [hunk 1]
```diff
@@ -15,16 +15,25 @@
...
     Core::ArgsParser args_parser;
+    args_parser.add_option(listen_address, "IP address to listen on", "listen-address", 'l', "listen_address");
     args_parser.add_option(port, "Port to listen on", "port", 'p', "port");
     args_parser.add_positional_argument(root_path, "Path to serve the contents of", "path", Core::ArgsParser::Required::No);
     args_parser.parse(argc, argv);
 
+    auto ipv4_address = IPv4Address::from_string(listen_address);
+    if (!ipv4_address.has_value()) {
+        printf("Warning: invalid listen address: %s\n", listen_address.characters());
+        ipv4_address = IPv4Address::from_string(default_listen_address);
```
Comment (linusg) [inline Userland/Services/WebServer/main.cpp:34]: That seems "dangerous" - passing an invalid IPv4 address should not make the webserver listen on 0.0.0.0, just print to stderr instead and exit.

```suggestion
        warnln("Invalid listen address: {}", listen_address);
        return 1;
```

**Review by linusg — CHANGES_REQUESTED** [review]
Nice! One behavioural change please, otherwise LGTM.

File: Userland/Services/WebServer/main.cpp (hunk 1 above)
Comment (EdwinHoksberg) [reply Userland/Services/WebServer/main.cpp]: Yes good one, I fixed it.

Comment (linusg) [reply Userland/Services/WebServer/main.cpp]: I just noticed we do the same for the port, I'll change that separately. Programs trying to be "helpful" that way isn't helping at all, IMO :D

**Pull request merged**